src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -3775,57 +3780,43 @@ InvalidateMetadataSystemCache(void)
 
 
 /*
- * DistTableOidList iterates over the pg_dist_partition table and returns
- * a list that consists of the logicalrelids.
+ * CitusTableIdList is a wrapper around CitusTableIdListByPredFunc that
+ * returns list OID's for citus tables.
  */
 List *
-DistTableOidList(void)
+CitusTableIdList(void)
 {
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 0;
-	List *distTableOidList = NIL;
-
-	Relation pgDistPartition = table_open(DistPartitionRelationId(), AccessShareLock);
-
-	SysScanDesc scanDescriptor = systable_beginscan(pgDistPartition,
-													InvalidOid, false,
-													NULL, scanKeyCount, scanKey);
-
-	TupleDesc tupleDescriptor = RelationGetDescr(pgDistPartition);
-
-	HeapTuple heapTuple = systable_getnext(scanDescriptor);
-	while (HeapTupleIsValid(heapTuple))
-	{
-		bool isNull = false;
-		Datum relationIdDatum = heap_getattr(heapTuple,
-											 Anum_pg_dist_partition_logicalrelid,
-											 tupleDescriptor, &isNull);
-		Oid relationId = DatumGetObjectId(relationIdDatum);
-		distTableOidList = lappend_oid(distTableOidList, relationId);
-
-		heapTuple = systable_getnext(scanDescriptor);
-	}
+	return CitusTableIdListByPredFunc(NULL);
+}
 
-	systable_endscan(scanDescriptor);
-	table_close(pgDistPartition, AccessShareLock);
 
-	return distTableOidList;
+/*
+ * CitusTableIdList is a wrapper around CitusTableIdListByPredFunc that
+ * returns list OID's for reference tables.
+ */
+List *
+ReferenceTableIdList(void)
+{
+	return CitusTableIdListByPredFunc(IsReferenceTableByDistParams);
 }
 
 
 /*
- * ReferenceTableOidList function scans pg_dist_partition to create a list of all
- * reference tables. To create the list, it performs sequential scan. Since it is not
- * expected that this function will be called frequently, it is OK not to use index scan.
- * If this function becomes performance bottleneck, it is possible to modify this function
- * to perform index scan.
+ * CitusTableIdListByPredFunc function scans pg_dist_partition and returns a
+ * list of OID's for the tables matching predicate function if specified.
+ * If predicate function is NULL, then this function returns OID's for all
+ * citus tables.
+ * To create the list, it performs sequential scan. Since it is not expected
+ * that this function will be called frequently, it is OK not to use index
+ * scan. If this function becomes performance bottleneck, it is possible to
+ * modify this function to perform index scan.
  */
 List *
-ReferenceTableOidList()
+CitusTableIdListByPredFunc(CheckDistParamsFunc checkDistParamsFunc)
```

Comment:
I think here as the parameter we can have `CitusTableType` to make this more clear and to avoid relying on dist params.

In the body of the function, we can use `IsCitusTableType` utility method.

Moreover we can rename this function as `CitusTableTypeIdList` to be also compatible with the existent utility functions such as `IsCitusTableType`

For example in `ReferenceTableIdList` it can be:

```
List *
ReferenceTableIdList(void)
{
	return CitusTableTypeIdList(REFERENCE_TABLE);
}
```

We can even remove `ReferenceTableIdList` and directly use `CitusTableTypeIdList(REFERENCE_TABLE);` in the places where it is called, however I am also fine with having `ReferenceTableIdList`.